tools/sqlite3_api_wrapper/sqlite3_udf_api/include/udf_struct_sqlite3.h

```diff
@@ -0,0 +1,29 @@
+#pragma once
+
+#include "duckdb/common/types.hpp"
+
+using namespace duckdb;
+
+struct sqlite3_value {
+    union MemValue {
+        double r;           /* Real value used when MEM_Real is set in flags */
+        int64_t i;          /* Integer value used when MEM_Int is set in flags */
+        int nZero;          /* Extra zero bytes when MEM_Zero and MEM_Blob set */
+        const char *zPType; /* Pointer type when MEM_Term|MEM_Subtype|MEM_Null */
+    } u;
+    // uint16_t flags;
+    PhysicalType type;
```

Comment:
Rather than using the physical type here, we can use a different enum. Note that the types that need to be stored in here are rather limited. If we look at the [sqlite3_value_type function](http://sqlite.org/c3ref/value_blob.html), we see the following types:

```c
SQLITE_INTEGER, SQLITE_FLOAT, SQLITE_TEXT, SQLITE_BLOB, or SQLITE_NULL.
```

This corresponds with the methods used for extracting the actual data value from the sqlite3_value wrapper:

```c
const void *sqlite3_value_blob(sqlite3_value*);
double sqlite3_value_double(sqlite3_value*);
int sqlite3_value_int(sqlite3_value*);
sqlite3_int64 sqlite3_value_int64(sqlite3_value*);
void *sqlite3_value_pointer(sqlite3_value*, const char*);
const unsigned char *sqlite3_value_text(sqlite3_value*);
```

In our sqlite3_value, I would store the same types using the same type enum (`int type`).